test/blackbox-tests/test-cases/special_files.t/run.t

```diff
@@ -6,7 +6,7 @@
 
   $ ln -s broken-link src/foo
   $ dune build @all
-  File "src/dune", line 1, characters 0-0:
+  File "src/foo", line 1, characters 0-0:
```

Comment:
I don't understand this test, so unsure if the error message is an improvement. Perhaps, it's worth adding a comment clarifying the purpose of this test.

Reply:
Hey, I don't understand it either 😅

After the changes to the file tree, `Rule.loc` gained a `Memo.Build.t`, which was problematic. So I changed the way the location of rules is calculated, which changed a few error messages. Some became worse, so I added a few missing `~loc` argument to calls to `Super_context.add_rule`.

But yh, since I'm editing this test let me try to understand it and improve it.

Reply:
Oh, it's special files in the sense symlinks/fifo/..., i.e. not a "regular file"

Reply:
Added a comment. @aalekseyev (git blame points to you), I don't understand what the last 3 lines of the test are for.